design-guidelines/interaction-and-feedback.md

```diff
@@ -0,0 +1,59 @@
+# Interaction and Feedback
+
+We can make our apps more intuitive/even easier to use by communicating the status of an interactive element, and providing.
+
+## Common interaction subjects
+
+- Actions
+  - For example: buttons
+- Selectable items
+  - For example: items in a list of selectable options
+- Data inputs
+  - For example: a text input
+
+## States
+
+### Disabled
+
+Use a disabled state when an element cannot be selected or interacted with, but there is something that can be done to enable the element. If there is nothing a user can do to enable an element, hide the element (see "Hidden" state for more info).
+
+#### When to disable an element
+
+An interactive element should be disabled when:
+
+- there is something a user can do in their current session to enable the element
+- something in progress needs to be completed before the element can be enabled
+- hiding the element might confuse users
+- we want to show that a feature that is on a product's roadmap. For example, disabling a button to select a cloud provider we don't support yet, but we want users to know that we plan to support it in the future.
+
+#### Make it clear why something is disabled
+
+If an element is disabled but still visible, don't force users to interpret why the element is disabled. Provide contextual information to explain why the element cannot be interacted with. There are no rules about how contextual information must be displayed, but two common patterns are: displaying caption text, and displaying tooltip when hovering or focusing the disabled element.
+
+#### Interaction guidelines
+
+It should be obvious that something is disabled before a user tries to interact with it.
+
+Disabled elements should not do provide any feedback when they're clicked.
+
+Disabled elements should not be keyboard-navigable unless a tooltip is displayed when the disabled element is focus.
+
+### Hovered/Focused
+
+TBD
+
+### Active/Selected
+
+TBD
+
+### Loading
+
+TBD
+
+### Read-only
+
+TBD
+
+### Hidden
+
+Hide an element when it cannot be selected or interacted with, and there is nothing a user can do to enable the element. Hiding elements instead of disabling the element helps to reduce cognitive lode because the user has less actions on the screen to think about.
```

Comment:
An element should be hidden: 
- When there is no other action a user can take during this session that would enable this action.
- When a user has been explicitly *not* granted permission to do an action (e.g., Create and Edit) and there is no expectation from the user that they would be able to take such an action (e.,g Create/Edit actions should be hidden from viewer roles.)
- An actions button or menu should be hidden when all actions are not available for the user because of permissions.